hnn_core/tests/test_cell_response.py

```diff
@@ -25,8 +25,37 @@ def test_cell_response(tmpdir):
                                  spike_types=spike_types,
                                  times=sim_times)
     cell_response.plot_spikes_hist(show=False)
-    cell_response.write(tmpdir.join('spk_%d.txt'))
+
+    # Testing writing using txt files
+    with pytest.warns(DeprecationWarning,
+                      match="Writing cell response to txt files is "
+                      "deprecated"):
+        cell_response.write(tmpdir.join('spk_%d.txt'))
+
+    # Testing reading from txt files
     assert cell_response == read_spikes(tmpdir.join('spk_*.txt'))
+
+    # Testing write using hdf5
+    cell_response.write(tmpdir.join('spk.hdf5'))
+
+    # Testing when overwrite is False and same filename is used
+    with pytest.raises(FileExistsError,
+                       match="File already exists at path "):
+        cell_response.write(tmpdir.join('spk.hdf5'), overwrite=False)
+
+    # Testing for wrong extension provided
+    with pytest.raises(NameError,
+                       match="File extension should be either txt or hdf5"):
+        cell_response.write(tmpdir.join('spk.xls'))
+
+    # Test read using hdf5
+    assert cell_response == read_spikes(tmpdir.join('spk.hdf5'))
+
+    # Testing File Not Found Error
+    with pytest.raises(FileNotFoundError,
+                       match="File not found at "):
+        assert cell_response == read_spikes(tmpdir.join('spk1.hdf5'))
```

Comment:
can you add a smoke test to check the visualization function works after you read the spike?

Reply:
> By visualization function do you mean plot_spikes_raster() and plot_spikes_hist() functions?

yep !

> Also I think I shouldn't have asserted anything here because I am checking for an error. Should I just call the function read_spikes() instead?

yes, that sounds more correct !